sdks/universal-router-sdk/src/utils/routerCommands.ts

```diff
@@ -30,17 +30,35 @@ export enum CommandType {
   EXECUTE_SUB_PLAN = 0x21,
 }
 
+export enum AltParser {
+  V4Call,
+}
+
 export enum Subparser {
   V3PathExactIn,
   V3PathExactOut,
 }
 
+export enum Parser {
+  Abi,
+  V4Actions,
+}
+
 export type ParamType = {
   readonly name: string
   readonly type: string
   readonly subparser?: Subparser
 }
 
+export type CommandDefinition =
+  | {
+      parser: Parser.Abi
+      params: ParamType[]
+    }
+  | {
+      parser: Parser.V4Actions
+    }
```

Comment:
ok, I've done a bit of a refactor here because we previously were not describing what was actually happening. 

Universal Router command inputs are _arbitrary bytes_, that can mean anything but currently are either:
- ABI encoded parameters for some action
- direct calldata for a function on the V4 router or the V4 Position manager

Previously we were just describing the ABI, which was insufficient for actually expressing what was happening in the v4 case. This is why we had a hard-coded edge case in `createCommand`:
```
  if (type === CommandType.V4_SWAP) {
    return { type, encodedInput: parameters[0] }
  }
```

I've expanded the command inputs definition to capture this nuance, allowing different top level "parsers" for each command type's inputs

Reply:
cc @hensha256

Reply:
ohhh interesting interesting, have you merged main? diana made some changes to that logic just this afternoon so want to make sure thats taken into account. But from what I can see youve already included that

Reply:
yea I did! I made a separate `V3Actions` vs `V4Actions`, where both bypass the abi decoding